tensorflow_recommenders_addons/dynamic_embedding/python/keras/layers/embedding_test.py

```diff
@@ -174,6 +178,58 @@ def test_backward_bp_v2(self):
       model.fit(x, y, verbose=0)
       self.assertAllEqual(emb_layer.params.size(), start)
 
+  def test_keras_save_load_weights(self):
+    if not context.executing_eagerly():
+      self.skipTest('Only test in eager mode')
+    save_dir = os.path.join(self.get_temp_dir(), "save_restore")
+    save_path = os.path.join(tempfile.mkdtemp(prefix=save_dir), "hash")
+
+    def model_fn(table_device):
+      input_tensor = tf.keras.layers.Input(shape=(1,), dtype=tf.int64)
+      embedding_out = de.keras.layers.BasicEmbedding(
+          embedding_size=1,
+          key_dtype=tf.int64,
+          value_dtype=tf.float32,
+          initializer=tf.keras.initializers.RandomNormal(),
+          devices=table_device,
+          name='test_keras_save_restore',
+      )(input_tensor)
+      model = tf.keras.Model(inputs=input_tensor, outputs=embedding_out)
+      optimizer = tf.keras.optimizers.Adam(learning_rate=1E-4, amsgrad=False)
+      optimizer = de.DynamicEmbeddingOptimizer(optimizer)
+      model.compile(optimizer=optimizer)
+      return model
+
+    table_device_ = ['/device:CPU:0']
```

Comment:
It would be better to test 'GPU' when GPU is available.

Reply:
fix